src/lsstseries/analysis/light_curve.py

```diff
@@ -0,0 +1,49 @@
+from typing import Optional
+import numpy as np
+
+MIN_OBSERVATIONS_REQUIRED_FOR_SF = 3
```

Comment:
Can this be moved into the subclass and passed into `LightCurve`?

Reply:
Yep, good call. I'll move this out of the base class and into the `StructureFunctionLightCurve` class.

Reply:
I've moved this constant to the `StructureFunctionLightCurve` class, and pass it as a parameter back to the base class which will do the array length check. Thanks for the suggestion here.